### ISSUE TITLE
simple_wml::node_to_string can cause the document to become invalid

### ISSUE THREAD

**Issue by Pentarctagon** [wesnoth/wesnoth#5390]
I've not investigated thoroughly exactly what's going on, but adding:
`std::cout << simple_wml::node_to_string(join_lobby_response.root()) << std::endl;`
immediately after the line https://github.com/wesnoth/wesnoth/blob/67bef9198c2f8c32114a0dbde21911a94e2a1f92/src/server/wesnothd/server.cpp#L806 will print out the document on cout, but then the `join_lobby_response` fails to be sent and gives the error:
```
20201226 13:39:57 error config: ERROR: 'unterminated element'
20201226 13:39:57 warning config: async_send_doc: simple_wml error: unterminated element
```

Just from the comment left at https://github.com/wesnoth/wesnoth/blob/67bef9198c2f8c32114a0dbde21911a94e2a1f92/src/server/common/simple_wml.cpp#L793-L803 though, I suspect that it's not always a correct statement.

**Comment by stevecotton** [comment]
I haven't tested it, but think output() is doing some "interesting memory compression", with a heap use after free:
* copying data into `buf`, which is the caller's temporary buffer `v`
* on lines 757 and 762, freeing the memory that its attributes were using and making them string_spans pointing into `buf`

https://github.com/wesnoth/wesnoth/blob/67bef9198c2f8c32114a0dbde21911a94e2a1f92/src/server/common/simple_wml.cpp#L755-L766

**Comment by soliton-** [comment]
This should be fixed with #9159.
